consensus/poet/core/tests/test_consensus/test_poet_block_publisher.py

```diff
@@ -0,0 +1,676 @@
+
+        # create a mock_consensus_state that returns a mock with
+        # the following settings:
+        mock_state = mock.Mock()
+        mock_state.validator_signup_was_committed_too_late.return_value = False
+        mock_state.validator_has_claimed_block_limit.return_value = True
+        mock_state.validator_is_claiming_too_early.return_value = False
+        mock_state.validator_is_claiming_too_frequently.return_value = False
+
+        mock_consensus_state.consensus_state_for_block_id.return_value = \
+            mock_state
+
+        mock_consensus_state_store.return_value.__getitem__.return_value = \
+            mock_consensus_state
+
+        # create mock_signup_info
+        mock_signup_info.unseal_signup_data.return_value = \
+            '00112233445566778899aabbccddeeff'
+
+        # create mock_batch_publisher
+        mock_batch_publisher = mock.Mock(
+            identity_signing_key=signing.generate_privkey())
+
+        mock_block_cache = mock.MagicMock()
+        mock_state_view_factory = mock.Mock()
+
+        # create mock_block_header with the following fields
+        mock_block = mock.Mock(identifier='0123456789abcdefedcba9876543210')
+        mock_block.header.signer_pubkey = '90834587139405781349807435098745'
+        mock_block.header.previous_block_id = '2'
+        mock_block.header.block_num = 1
+        mock_block.header.state_root_hash = '6'
+        mock_block.header.batch_ids = '4'
+
+        # check test
+        with mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.'
+                        'LOGGER') as mock_logger:
+            block_publisher = \
+                poet_block_publisher.PoetBlockPublisher(
+                    block_cache=mock_block_cache,
+                    state_view_factory=mock_state_view_factory,
+                    batch_publisher=mock_batch_publisher,
+                    data_dir=self._temp_dir,
+                    config_dir=self._temp_dir,
+                    validator_id='validator_deadbeef')
+
+            self.assertFalse(
+                block_publisher.initialize_block(
+                    block_header=mock_block.header))
+
+            # Could be a hack, but verify that the appropriate log message is
+            # generated - so we at least have some faith that the failure was
+            # because of what we are testing and not something else.  I know
+            # that this is fragile if the log message is changed, so would
+            # accept any suggestions on a better way to verify that the
+            # function fails for the reason we expect.
+
+            (message, *_), _ = mock_logger.error.call_args
+            self.assertTrue('Validator has reached maximum number of '
+                            'blocks with key pair' in message)
+
+            # check that create.signup_info() wasn't called
+            # to renew the set of keys
+            self.assertFalse(mock_signup_info.create_signup_info.called)
```

Comment:
There should be an additional test that ensures that if the PoET block publisher has not refreshed the key that it does so (i.e., calls `create_signup_info`, etc.).